jsapp/js/components/modalForms/formMedia.es6

```diff
@@ -76,72 +110,41 @@ class FormMedia extends React.Component {
-          // back end checks for valid base64 encoded string first
-          } else if (backendErrorText === BAD_UPLOAD_MEDIA_ENCODING_ERROR) {
-            alertify.error(t('Your uploaded media does not contain base64 valid content. Please check your media content.'));
-          } else if (
-              backendErrorText === BAD_UPLOAD_MEDIA_TYPE_ERROR ||
-              backendErrorText === GENERIC_BAD_UPLOAD_MEDIA_ENCODING_ERROR
-            ) {
-              alertify.error(t('Your uploaded media does not contain one of our supported MIME filetypes: `image`, `audio`, `video`, `text/csv`, `application/xml`'));
-          } else if (backendErrorText === INTERNAL_SERVER_ERROR) {
-            alertify.error(t('File could not be uploaded!'));
-          }
-	    });
+        this.uploadMedia(formMediaJSON);
       });
     }
   }
 
-  onSubmitURL() {
-    var urlInputField = $(document).find('input.form-media__url-input').eq(0);
-    var url = urlInputField.val();
+  onInputURLChange(inputURL) {
+    this.setState({inputURL: inputURL});
+  }
 
-    // Clear the url field
-    urlInputField.val('');
+  onSubmitURL() {
+    var url = this.state.inputURL;
 
     if (url === '') {
       alertify.warning(t('URL is empty!'));
     } else {
-      this.setState({isUploadURLPending: true})
+      this.setState({
+        isUploadURLPending: true,
+        inputURL: ''
+      });
 
       var formMediaJSON = {
         description: 'default',
         file_type: 'form_media',
         metadata: JSON.stringify({redirect_url: url}),
       };
-      dataInterface.postFormMedia(this.props.asset.uid, formMediaJSON).done(() => {
-        this.refreshFormMedia();
-      }).fail((err) => {
-        this.setState({isUploadURLPending: false});
-
-        var backendErrorText = (err.responseJSON != undefined) ? err.responseJSON.metadata[0] : err.statusText;
-        if (backendErrorText === BAD_URL_ERROR) {
-          alertify.warning(t('The URL you entered is not valid'));
-        } else if (backendErrorText === FILE_ALREADY_EXISTS_URL_ERROR ) {
-          alertify.error(t('File already exists!'));
-        } else if (backendErrorText === BAD_URL_MEDIA_TYPE_ERROR) {
-          alertify.error(t('Your URL media does not contain one of our supported MIME filetypes: `image`, `audio`, `video`, `text/csv`, `application/xml`'));
-        } else if (backendErrorText === INTERNAL_SERVER_ERROR) {
-          alertify.error(t('Your URL media failed to upload!'));
-        }
-      });
+      this.uploadMedia(formMediaJSON);
     }
   }
 
-  removeMedia(url) {
-    dataInterface.deleteFormMedia(url).done(() => {
-      this.refreshFormMedia();
-    }).fail(() => {
-      alertify.error(t('Failed to delete media!'));
-    });
+  onDeleteMedia(url) {
+    const callbacks = {
+      onComplete: this.onGetMediaCompleted.bind(this),
+      onFail: this.onDeleteMediaFailed.bind(this)
+    };
+    actions.media.deleteMedia(this.props.asset.uid, url, callbacks);
```

Comment:
Similarly to above - remove `callbacks` and add listeners to `componentDidMount`:

```
actions.media.deleteMedia.completed.listen(this.onGetMediaCompleted);
actions.media.deleteMedia.failed.listen(this.onDeleteMediaFailed);
```

Reply:
First attempt had a lot of :spaghetti:! I think I learned from this for #2928 :crossed_fingers: